app/service/data_svc.py

```diff
@@ -293,6 +293,7 @@ def __init__(self, dao):
         for a in agents:
             executors = await self.dao.get('core_executor', criteria=dict(agent_id=a['id']))
             a['executors'] = [dict(executor=e['executor'], preferred=e['preferred']) for e in executors]
+            a['sleep'] = self.jitter('{}/{}'.format(a['sleep_min'], a['sleep_max']))
```

Comment:
why this? this change is in explode_agents, which gets called as a read-only function. this change sets a temporary sleep value for agents (not persisted)... right?

Reply:
oh I see what you're doing. you're doing this intentionally so that the handle_heartbeat function will always get a different sleep value.. creative- but this belongs in the handle_heartbeat function

Reply:
I pulled it out of the handle_heartbeat because you said it was too verbose!

Reply:
```
        if agent:
            update_data = dict(last_seen=now, pid=pid, ppid=ppid)
            if agent[0]['trusted']:
                update_data['last_trusted_seen'] = now
            await self.get_service('data_svc').update('core_agent', 'paw', paw, data=update_data)
        else:
            queued = dict(last_seen=now, paw=paw, platform=platform, server=server, host_group=group,
                          location=location, architecture=architecture, pid=pid, ppid=ppid,
                          trusted=True, last_trusted_seen=now, sleep_min=sleep, sleep_max=sleep)
            await self.get_service('data_svc').create_agent(agent=queued, executors=executors)
            await self.get_service('data_svc').explode_agents(criteria=dict(paw=paw))
        agent[0]['sleep'] = self.jitter('{}/{}'.format(agent[0]['sleep_min'], agent[0]['sleep_max']))
        return agent[0]
```
That's what I originally had essentially before I moved it to the data_svc